src/Form/ImportVideoItemByMpxId.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\media_mpx\Form;
+
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Logger\RfcLogLevel;
+use Drupal\Core\Utility\Error;
+use Drupal\media_mpx\Repository\MpxMediaType;
+use Drupal\media_mpx\Service\UpdateVideoItem\UpdateVideoItem;
+use Drupal\media_mpx\Service\UpdateVideoItem\UpdateVideoItemRequest;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Form class to import a single mpx item based on its id.
+ *
+ * @package Drupal\media_mpx\Form
+ */
+class ImportVideoItemByMpxId extends FormBase {
+
+  /**
+   * The Update Video Item service.
+   *
+   * @var \Drupal\media_mpx\Service\UpdateVideoItem\UpdateVideoItem
+   */
+  private $updateVideoItem;
+
+  /**
+   * The system logger.
+   *
+   * @var \Psr\Log\LoggerInterface
+   */
+  private $logger;
+
+  /**
+   * The media type repository.
+   *
+   * @var \Drupal\media_mpx\Repository\MpxMediaType
+   */
+  private $mpxTypeRepository;
+
+  /**
+   * UpdateMediaItemForAccount constructor.
+   */
+  public function __construct(UpdateVideoItem $updateVideoItem, MpxMediaType $mpxTypeRepository, LoggerInterface $logger) {
+    $this->updateVideoItem = $updateVideoItem;
+    $this->mpxTypeRepository = $mpxTypeRepository;
+    $this->logger = $logger;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('media_mpx.service.update_video_item'),
+      $container->get('media_mpx.repository.mpx_media_types'),
+      $container->get('logger.channel.media_mpx')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    if (!$video_opts = $this->loadVideoTypeOptions()) {
+      $this->messenger()->addError($this->t('There has been an unexpected problem loading the form. Reload the page.'));
```

Comment:
@salvamomo why could this happen? Just curious. Also, would reloading the page solve it?

Reply:
Rather unlikely, and not related this module itself (say db connection failing for a moment, which could break the whole site anyway and not only this page), but not impossible.

Reply:
This is fine for now, but a more robust method would be to throw an exception from `loadVideoTypeOptions()`. "object || false" returns are evil :D